easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialogHolderActivity.java

```diff
@@ -0,0 +1,40 @@
+package pub.devrel.easypermissions;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.RestrictTo;
+import android.support.v7.app.AppCompatActivity;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class AppSettingsDialogHolderActivity extends AppCompatActivity implements DialogInterface.OnClickListener {
```

Comment:
This is a very interesting way to go about the whole thing.

For the rationale dialog we went with DialogFragment which can survive rotation, however the crash in #65 is a result of using DialogFragment in onActivityResult.

I think because of how bad that crash is, your approach is likely better.